S3/S3.py

```diff
@@ -707,7 +711,7 @@ def object_modify(self, src_uri, dst_uri, extra_headers = None):
         headers = self._sanitize_headers(headers)
         acl = self.get_acl(src_uri)
 
-        headers['x-amz-copy-source'] = "/%s/%s" % (src_uri.bucket(), self.urlencode_string(src_uri.object()))
+        headers['x-amz-copy-source'] = deunicodise("/%s/%s" % (src_uri.bucket(), self.urlencode_string(src_uri.object())), encoding="UTF-8")
```

Comment:
You are right about that one. I just did the two in 2 different times, so was not consistent.
I just pushed an additional commit to fix that and use deunicodise in both case.
I prefer using deunicodise because there is the failsafe check for string being unicode.

I was a little lazy at that time, but I think for a future patch to add functions like:
s3_encode / s3_decode  or deunicodise_s3 or unicodise_s3
and to use that instead of the "encode /deunicodise" (...=UTF-8) that are used throughout the code to prepare string to be sent to the S3 server.